### PR TITLE
shared | add EnvVar to disable telemetry

### PR DESCRIPTION
**Pull request by ferpizza**
### Description
Describe the proposed changes made in this PR.

### How was this PR tested?
Describe how you tested this PR.

### CODE REVIEW

**Review by jotare — APPROVED** [review]

File: charts/nucliadb_shared/templates/nucliadb.cm.yaml [hunk 1]
```diff
@@ -87,4 +87,8 @@ data:
   JAEGER_QUERY_HOST: "{{ .Values.tracing.jaeger_query_host }}"
   JAEGER_QUERY_PORT: "{{ .Values.tracing.jaeger_query_http_port }}"
   JAEGER_PROPAGATION: "jaeger,b3,w3c"
-  OTEL_EXPORTER_JAEGER_AGENT_SPLIT_OVERSIZED_BATCHES: "1"
\ No newline at end of file
+  OTEL_EXPORTER_JAEGER_AGENT_SPLIT_OVERSIZED_BATCHES: "1"
+
+{{- if not .Values.telemetry.enabled }}
+  nucliadb_DISABLE_TELEMETRY: "True"
```
Comment (alekece) [inline charts/nucliadb_shared/templates/nucliadb.cm.yaml:93]: What is the reason behind mixing lower and upper characters in variable name ?

Comment (ferpizza) [reply charts/nucliadb_shared/templates/nucliadb.cm.yaml:93]: that's how the variable is declard in code ... I'm just adding it to helm charts to be able to set it on deploy

Comment (ferpizza) [reply charts/nucliadb_shared/templates/nucliadb.cm.yaml:93]: this is the relevant line of code
https://github.com/nuclia/nucliadb/blob/main/nucliadb_telemetry/src/lib.rs#L57

Comment (hermeGarcia) [reply charts/nucliadb_shared/templates/nucliadb.cm.yaml]: Fixed at [6d41257](https://github.com/nuclia/nucliadb/pull/515/commits/6d41257f6e0437b52203a785e85e1cdccb3990c9)

**Review by alekece — APPROVED** [review]

**Review by jotare — CHANGES_REQUESTED** [review]
Upper case nucliadb_DISABLE_TELEMETRY in docs (docs/reference/telemetry.md)

File: nucliadb/nucliadb/search/tests/node.py [hunk 2]
```diff
@@ -40,6 +40,7 @@
         "VECTORS_DIMENSION": "768",
         "DATA_PATH": "/data",
         "READER_LISTEN_ADDRESS": "0.0.0.0:4445",
+        "NUCLIADB_DISABLE_TELEMETRY": "True",
```
Comment (jotare) [inline nucliadb/nucliadb/search/tests/node.py:43]: Does it matter which value the env var have?

**Review by jotare — APPROVED** [review]

**Pull request merged**